SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRPCTimeout/SpatialTestRPCTimeout.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestRPCTimeout.h"
+#include "Components/BoxComponent.h"
+#include "GameFramework/Character.h"
+#include "GameFramework/PlayerController.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialTestRPCTimeoutGameMode.h"
+#include "SpatialTestRPCTimeoutPlayerController.h"
+
+/**
+ * This test ensures that RPC calls with unresolved parameters are queued until their parameters are correctly resolved.
+ * This test contains 1 Server and 2 Client workers running in multiple processes.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - Launch at least one client in a separate process
+ * - Test:
+ *  - All clients launched outside of the editor process must ensure that the referenced material asset is not in memory at the start of the
+ * test.
+ *  - All clients launched outside of the editor process must successfully resolve the material asset before passing it into a client RPC
+ * function.
+ */
+
+ASpatialTestRPCTimeout::ASpatialTestRPCTimeout()
+	: Super()
+{
+	Author = "Iwan";
+	Description = TEXT(
+		"This test calls an RPC with an asset that was softly referenced and check that it will be asynchronously loaded with a timeout of "
+		"0.");
+}
+
+void ASpatialTestRPCTimeout::CreateCustomContentForMap()
+{
+	GetWorldSettingsForMap()->DefaultGameMode = ASpatialTestRPCTimeoutGameMode::StaticClass();
```

Comment:
I think you can move all the things from the rpc map here.

Reply:
Ah just noticed this, would that be better than putting everything in the map source files?

Reply:
It would allow you to remove the map cpp and h files, and just have everything here
your call on whether it's needed.

Reply:
Got it, will do that today